content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -751,26 +751,154 @@ with the response:
 
 ```
 The CronTab "my-new-cron-object" is invalid:
-* spec: Invalid value: map[string]interface {}{"minReplicas": 0, "replicas":20, "maxReplicas": 10}: failed rule: self.minReplicas <= self.replicas && self.replicas <= self.maxReplicas
-```
-
-TODO: (using text from types_jsonprops.go and KEP were applicable, but using "full" multi-line examples that include both the schema and the custom resource data)
-- Explain that rules are compiled when CRDs are created/updated. Show full example including compilation error output examples.
-- Explain scope, self, and how objects, maps and arrays are accessed. Show full examples.
-  - Must show: 'self.field' selection, has() field presence checking, 'self[key]' map access and
-    'key in self' map containment, 'self[i]' list access, all/exists/filter and how they apply to
-    maps and lists. Show more of the functions than covered in types_jsonprops.go. Provide links to
-    functions and macros in spec.  Provide link to strings extension library in cel-go that we have
-    enabled. Explain that this is an extension library.
-- Include examples table from KEP? Probably just past it in and add context.
-- Explain access to type and object meta.
-- Explain openapiv3 -> CEL declarations type mapping and include the table from the KEP. Link to OpenAPIv3 and CEL documentation about types.
-- Explain int-or-string, preserve-unknown, nullable, embedded. Provide some short examples.
-- Explain escaping using a table. Provide some short examples. Provide guidance on how to name
-  properties (both in this section of this document and elsewhere in this document where property
-  names are introduced/discussed).
-- Explain + and == for list maps and list sets (table? whatever looks better)
-- DO NOT: provide all the motivation and design rationale from the KEP.
+* spec: Invalid value: map[string]interface {}{"maxReplicas":10, "minReplicas":0, "replicas":20}: failed rule: self.replicas <= self.maxReplicas
+```
+
+`x-kubernetes-validations` could have multiple rules. 
+
+The `rule` under `x-kubernetes-validations` represents the expression which will be evaluated by CEL.
+
+The `message` represents the message displayed when validation fails. The message is required if the Rule contains
+line breaks. The message must not contain line breaks. If unset, the message is "failed rule: {Rule}".
```

Comment:
How about a multi-line example showing multiple rules with messages?

Reply:
Also, I don't think we need to be quite as detailed about the exact rules of the message field here where we first introduce it.  I'd say only what it does here and then teach with an example.  It's okay down at the end of the doc somewhere to spell out the exact rules if you want.

Reply:
I have added message into example and add the response example with message unset.

Reply:
This part of the above comment is still unaddressed right? "I don't think we need to be quite as detailed about the exact rules of the message field here where we first introduce it. I'd say only what it does here and then teach with an example. It's okay down at the end of the doc somewhere to spell out the exact rules if you want."

Reply:
> I'd say only what it does here and then teach with an example. 
I thought we are ok with what it has now. I will reduce the details on message.